core/transaction_watcher.md

```diff
@@ -0,0 +1,38 @@
+## TransactionWatcher
+
+TransactionWatcher allows one to continuously watch (monitor), by means of polling, the status of a given transaction.
+
+We need a 'TransactionFetcher` that fetches the transaction from the network.
+
+```
+interface TransactionFetcher:
+    get_transaction(tx_hash: string): TransactionOnNetwork;
```

Comment:
`TransactionOnNetwork` isn't yet defined in the specs, but that isn't necessarily an issue, since it will soon be defined.